exp/operator/controllers/coreprovider_controller.go

```diff
@@ -30,17 +29,15 @@ import (
 // CoreProviderReconciler reconciles a CoreProvider object
 type CoreProviderReconciler struct {
 	client.Client
-	Log    logr.Logger
 	Scheme *runtime.Scheme
 }
 
 // +kubebuilder:rbac:groups=operator.cluster.x-k8s.io,resources=coreproviders,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=operator.cluster.x-k8s.io,resources=coreproviders/status,verbs=get;update;patch
 
-func (r *CoreProviderReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("coreprovider", req.NamespacedName)
-
+func (r *CoreProviderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	_ = log
```

Comment:
```suggestion
```

I am not sure why we need the logger in this PR. Leaving it as a noop seems fine to me.

Reply:
+1 to drop if not required

Reply:
done